compiler/plugins/input/TOSA/InputConversion/test/tosa.pdll

```diff
@@ -0,0 +1,112 @@
+
+// declare native rewrite called "rewriteAsFlowDispatch"
+Rewrite rewriteAsFlowDispatch(
+        op : Op, fn_name : Attr, 
+        inputs : ValueRange, 
+        repl_vals : ValueRange, 
+        repl_dims : ValueRange, 
+        other : ValueRange);
+
+
+Constraint checkElementType(t1 : Type, t2 : Type);
+
+Pattern mlp_tosa with benefit(1){
+  let elemTy = type<"f32">;
+
+  let matMulOp = op<tosa.matmul>(lhs : Value<LhsTy : Type>, rhs : Value<RhsTy : Type>) -> (MatMulTy : Type);
+  checkElementType(LhsTy, elemTy);
+  checkElementType(RhsTy, elemTy);
+  checkElementType(MatMulTy, elemTy);
+
+  let reluOp = op<tosa.clamp>(matMulOp.0) {
+    min_int = attr<"0 : i64">,
+    max_int = maxIntValue: Attr,
+    min_fp = attr<"0.000000e+00 : f32">,
+    max_fp = maxFloatValue: Attr
+  } -> (ReluTy : Type);
+
+  rewrite matMulOp with {
+    let IntTy = type<"i32">;
+    let IndexTy = type<"index">;
+    let twoOp = op<arith.constant> { value = attr<"2 : index"> } -> (IndexTy);
+    let oneOp = op<arith.constant> { value = attr<"1 : index"> } -> (IndexTy);
+    let m = op<tensor.dim>(lhs, oneOp) -> (IndexTy);
+    let n = op<tensor.dim>(rhs, twoOp)  -> (IndexTy);
+    let k = op<tensor.dim>(lhs, twoOp); 
+    let m_cast = op<arith.index_cast>(m)  -> (IntTy);
+    let n_cast = op<arith.index_cast>(n) -> (IntTy);
+    let k_cast = op<arith.index_cast>(k) -> (IntTy);
+
+
+
+    let doRelu = op<arith.constant> { value = attr<"1 : i1">};
+    let inputs = (lhs, rhs);
+    let repl_vals = (reluOp.0 );
+    let repl_dims = ();
+    let other = (m_cast.0, n_cast.0, k_cast.0, doRelu.0);
+
+
+    
+    // The `rewriteAsFlowDispatch` is a rewrite function that allows
+    // converting the matched dag into a call to the external function call
+    // provided by a system plugin. The rewrite method expects the following
+    // arguments
+    // - the root of the matched DAG. This op will be erased after the call.
+    // - `fn_name` the name of the function that is provided externally
+    //   (using a plugin).
+    // - `input_values` are values that are captures as the part of the match
+    //   and are inputs to the match.
+    // - `replaced_values` are the values that are captured as part of the
+    //   match and are replaced by the `flow.dispatch`. The `flow.dispatch`
+    //   returns as many values as `replaced_values` (and of same type).
+    // - `replaced_values_dims` are the values for the dynamic dimensions of
+    //   all the `tensor` values in `replaced_values`. For matches that could
+    //   be static or dynamic, it should be assumed that the shape is dynamic
+    //   and the value needs to be passed to the rewrite function.
+    // - `other_operands` same as `input_values`, but kept separate to allow
+    //   flexibility of where the results are passed through the ABI boundary.
+    rewriteAsFlowDispatch(
+      reluOp, attr<"\"mlp_external\"">, 
+      inputs, repl_vals, 
+      repl_dims, other);
+  }; 
```

Comment:
I see almost this exact spec, in particular with all of the same docs in `torch.pdll`, `mlp_linalg_spec.pdll`, `mlp_torch_spec.pdll`, and `mlp_tosa_spec.pdll`. This is a fairly large spec file to use to just show proof of life for different kinds of inputs. Can we deduplicate these a bit?